src/hb-ot-layout.cc

```diff
@@ -1009,43 +1027,109 @@ hb_ot_layout_get_size_params (hb_face_t    *face,
 			      unsigned int *range_start,       /* OUT.  May be nullptr */
 			      unsigned int *range_end          /* OUT.  May be nullptr */)
 {
-  const OT::GPOS &gpos = _get_gpos (face);
-  const hb_tag_t tag = HB_TAG ('s','i','z','e');
-
-  unsigned int num_features = gpos.get_feature_count ();
-  for (unsigned int i = 0; i < num_features; i++)
+  hb_tag_t tag = HB_TAG ('s','i','z','e');
+  const OT::FeatureParams &feature_params =
+    _get_gsubgpos_matched_feature_params (face, tag);
+  if (&feature_params != &OT::Null (OT::FeatureParams))
   {
-    if (tag == gpos.get_feature_tag (i))
+    const OT::FeatureParamsSize params = feature_params.get_size_params (tag);
+    if (params.designSize)
     {
-      const OT::Feature &f = gpos.get_feature (i);
-      const OT::FeatureParamsSize &params = f.get_feature_params ().get_size_params (tag);
-
-      if (params.designSize)
-      {
 #define PARAM(a, A) if (a) *a = params.A
-	PARAM (design_size, designSize);
-	PARAM (subfamily_id, subfamilyID);
-	PARAM (subfamily_name_id, subfamilyNameID);
-	PARAM (range_start, rangeStart);
-	PARAM (range_end, rangeEnd);
+      PARAM (design_size, designSize);
+      PARAM (subfamily_id, subfamilyID);
+      PARAM (subfamily_name_id, subfamilyNameID);
+      PARAM (range_start, rangeStart);
+      PARAM (range_end, rangeEnd);
 #undef PARAM
 
-	return true;
-      }
+      return true;
     }
   }
 
-#define PARAM(a, A) if (a) *a = 0
-  PARAM (design_size, designSize);
-  PARAM (subfamily_id, subfamilyID);
-  PARAM (subfamily_name_id, subfamilyNameID);
-  PARAM (range_start, rangeStart);
-  PARAM (range_end, rangeEnd);
-#undef PARAM
+#define UNPARAM(a) if (a) *a = 0
+  UNPARAM (design_size);
+  UNPARAM (subfamily_id);
+  UNPARAM (subfamily_name_id);
+  UNPARAM (range_start);
+  UNPARAM (range_end);
+#undef UNPARAM
 
   return false;
 }
 
+/**
+ * hb_ot_layout_get_feature_params:
+ *
+ * Returns bunch of NameIDs and other related information for a tag,
+ * and sets NameIDs parameters (suffixed with _id) to 0xFFFF in case
+ * the requested tag not found or the tag doesn't support it.
+ *
+ * Since: REPLACEME
```

Comment:
Please document the function arguments as well, otherwise gi-introspection will be broken.